dynamiqs/options.py

```diff
@@ -41,13 +39,15 @@ def __init__(
             dtype (torch.dtype, optional): Complex data type to which all complex-valued
                 tensors are converted. `t_save` is also converted to a real data type of
                 the corresponding precision.
-            device (torch.device, optional): Device on which the tensors are stored.
+            device (string or torch.device, optional): Device on which the tensors are
+                stored.
         """
         self.gradient_alg = gradient_alg
         self.save_states = save_states
         self.verbose = verbose
-        self.dtype = dtype
-        self.device = device
+        self.cdtype = to_cdtype(dtype)
+        self.rdtype = dtype_complex_to_real(self.cdtype)
```

Comment:
Wait why do we have two different functions `to_cdtype` and `dtype_real_to_complex` again? Shouldn't we merge them?